explorer-api/app/Explorer/Web/Validate/Random.hs

```diff
@@ -45,70 +39,54 @@ import Database.Esqueleto
-        pure $ maybeToEither errMsg unValue (listToMaybe res1)
+    res <- select . from $ \ txOut -> do
+             firstRandomRow
+             pure (txOut ^. TxOutAddress)
+    pure $ maybeToEither errMsg unValue (listToMaybe res)
   where
     errMsg :: LookupFail
-    errMsg = DbLookupMessage "queryRandomAddress: Lookup address by index failed"
+    errMsg = DbLookupMessage "queryRandomAddress: Lookup address"
 
 queryRandomBlockHash :: MonadIO m => ReaderT SqlBackend m (Either LookupFail ByteString)
 queryRandomBlockHash = do
     res <- select . from $ \ blk -> do
-              where_ (blk ^. BlockTxCount >. val 0)
-              pure countRows
-    case listToMaybe res of
-      Nothing -> pure $ Left (DbLookupMessage "queryRandomBlockHash: Empty Block table")
-      Just (Value blkCount) -> do
-        blkid <- liftIO $ randomRIO (1, blkCount - 1)
-        res1 <- select . from $ \ blk -> do
-                  where_ (blk ^. BlockTxCount >. val 0)
-                  orderBy [asc (blk ^. BlockId)]
-                  offset blkid
-                  limit 1
-                  pure (blk ^. BlockHash)
-        pure $ maybeToEither errMsg unValue (listToMaybe res1)
+             where_ (blk ^. BlockTxCount >. val 0)
+             firstRandomRow
+             limit 1
+             pure (blk ^. BlockHash)
+    pure $ maybeToEither errMsg unValue (listToMaybe res)
   where
     errMsg :: LookupFail
     errMsg = DbLookupMessage "queryRandomBlockHash: Lookup block by index failed"
 
 queryRandomRedeemAddress :: MonadIO m => ReaderT SqlBackend m (Either LookupFail Text)
 queryRandomRedeemAddress = do
     res <- select . from $ \ (tx `InnerJoin` txOut) -> do
-              on (tx ^. TxId ==. txOut ^. TxOutTxId)
-              -- Block 1 contains all the TxOuts from the Genesis Distribution.
-              where_ (tx ^. TxBlock ==. val (mkBlockId 1))
-              pure countRows
-    case listToMaybe res of
-      Nothing -> pure $ Left (DbLookupMessage "queryRandomAddress: Empty TxOut table")
-      Just (Value txoCount) -> do
-        txoid <- liftIO $ randomRIO (1, txoCount - 1)
-        res1 <- select . from $ \ txOut -> do
-                  where_ (txOut ^. TxOutId ==. val (mkTxOutId txoid))
-                  pure (txOut ^. TxOutAddress)
-        pure $ maybeToEither errMsg unValue (listToMaybe res1)
+             -- Block 1 contains all the TxOuts from the Genesis Distribution.
+             where_ (tx ^. TxBlock ==. val (mkBlockId 1))
+             on (tx ^. TxId ==. txOut ^. TxOutTxId)
+             firstRandomRow
+             pure (txOut ^. TxOutAddress)
+    pure $ maybeToEither errMsg unValue (listToMaybe res)
   where
-    errMsg :: LookupFail
-    errMsg = DbLookupMessage "queryRandomRedeemAddress: Lookup address by index failed"
-
-mkTxOutId :: Word -> TxOutId
-mkTxOutId = TxOutKey . fromIntegral
+    errMsg ::  LookupFail
+    errMsg = DbLookupMessage "queryRandomRedeemAddress: Lookup address"
```

Comment:
Was this error message shortened on purpose?

Reply:
Ah thanks. The change was deliberate, but I intended to replace it with, "Lookup random X failed".

Fixed. 👍